### ISSUE TITLE
ElasticSearch 7.x is no healthy with recipes released

### ISSUE THREAD

**Issue by jvalderrama** [elastic/ansible-elasticsearch#561]
**Describe the feature**: ES 7.x completes installation but cluster is not healthy

**Elasticsearch version** 7.0.1

**Role version**:  (If using master please specify github sha)

**JVM version** (`java -version`): openjdk version "1.8.0_201"

**OS version** (`uname -a` if on a Unix-like system): Linux x86_64

**Description of the problem including expected versus actual behaviour**:

The ES cluster installation is completed correctly with the support includes for 7.x version but the resulting cluster is not healthy due these responses:

http://localhost:9200/_xpack **without license, this is null**

```
{
build: {
hash: "e4efcb5",
date: "2019-04-29T12:57:49.344766Z"
},
license: null,
features: {
ccr: {
description: "Cross Cluster Replication",
available: true,
enabled: true
},
graph: {
description: "Graph Data Exploration for the Elastic Stack",
available: true,
enabled: true
}, ....
```

**The nodes are unreachable and is not healthty**

http://localhost:9200/_cluster/health 
http://localhost:9200/_cat/nodes 
http://localhost:9200/_cat/health 

```
{
error: {
root_cause: [
{
type: "master_not_discovered_exception",
reason: null
}
],
type: "master_not_discovered_exception",
reason: null
},
status: 503
}
```

http://localhost:9200/ **goes fine:**

```
{ name: "localhost-master", cluster_name: "cluster-dual-v7", cluster_uuid: "_na_", version: { number: "7.0.1", build_flavor: "default", build_type: "rpm", build_hash: "e4efcb5", build_date: "2019-04-29T12:56:03.145736Z", build_snapshot: false, lucene_version: "8.0.0", minimum_wire_compatibility_version: "6.7.0", minimum_index_compatibility_version: "6.0.0-beta1" }, tagline: "You Know, for Search" }
```

**Playbook**:
```yaml
- hosts: master
  roles:
    - role: ansible-elasticsearch
  vars:
    es_enable_xpack: true
    es_instance_name: "master"
    es_config:
       network.host: "10.251.64.71"
       cluster.name: "cluster-dual-v7"
       discovery.zen.ping.unicast.hosts: "10.251.64.71:9301"
       http.port: 9200
       transport.tcp.port: 9301
       node.data: false
       node.master: true
       bootstrap.memory_lock: true
    es_scripts: false
    es_templates: false
    es_version_lock: false
    es_heap_size: 1g
    es_api_host: "10.251.64.71"
    es_api_port: 9200
    es_api_basic_auth_username: elastic
    es_api_basic_auth_password: changeme

- hosts: data
  roles:
    - role: ansible-elasticsearch
  vars:
    es_enable_xpack: true
    es_instance_name: "data"
    es_config:
       network.host: "10.251.64.72"
       cluster.name: "cluster-dual-v7"
       discovery.zen.ping.unicast.hosts: "10.251.64.71:9301"
       http.port: 9200
       transport.tcp.port: 9301
       node.data: true
       node.master: false
       bootstrap.memory_lock: true
    es_scripts: false
    es_templates: false
    es_version_lock: false
    es_heap_size: 1g
    es_api_host: "10.251.64.72"
    es_api_port: 9200
    es_api_basic_auth_username: elastic
    es_api_basic_auth_password: changeme
```

**Provide logs from Ansible**:

**ES Logs if relevant**:

**Comment by fatmcgav** [comment]
@jvalderrama Cheers for opening this issue. 

I believe your problem is related to the change in how Elasticsearch 7 forms clusters. More details on those changes can be found here: https://www.elastic.co/guide/en/elasticsearch/reference/current/discovery-settings.html

TL;DR:
I believe adding the following to the `vars` config for your `master` play should get you running:
```
    cluster.initial_master_nodes: 
      - 10.251.64.71
```

Also, for usage issues such as this, we tend to recommend using Discuss at https://discuss.elastic.co/c/elasticsearch. Otherwise issues can sometimes get missed or not responded to for a while :)

**Comment by jvalderrama** [comment]
@fatmcgav :+1: Thanks a lot for the response, I was able to complete a successfully installation and now it is working.
Also I am going to take in mind your suggestion about the usage issues. :)))

**Comment by xaviersavio** [comment]
I'm having same issues as License is showing null after upgrade from 6.8 to 7.3 on Windows Server 2012 R2

How we can update basic  license

**Comment by xaviersavio** [comment]
![Capture](https://user-images.githubusercontent.com/53714631/62989561-9a097f00-be9c-11e9-8c3f-9f23390abe09.JPG)

**Comment by jmlrt** [comment]
Hi @xaviersavio, this ansible role is currently not supported on Windows platform.

**Comment by xaviersavio** [comment]
It supports

Sent from Outlook
________________________________

**Comment by gspgsp** [comment]
I'm havaing the same issue,
{
"error": {
"root_cause": [
{
"type": "master_not_discovered_exception",
"reason": null
}
],
"type": "master_not_discovered_exception",
"reason": null
},
"status": 503
}

that's my config:

master:

cluster.name: my-clusters
node.name: node-2
node.master: true
#node.data: true
path.data: /data/cluster/elasticsearch
path.logs: /var/log/elasticsearch
network.host: 0.0.0.0
http.port: 9200
discovery.seed_hosts: ["172.16.218.199","172.16.107.151"]
cluster.initial_master_nodes: ["node-2"]


slave:

cluster.name: my-clusters
node.name: node-1
node.master: false
node.data: true

path.data: /data/cluster/elasticsearch

path.logs: /var/log/elasticsearch

network.host: 0.0.0.0
http.port: 9200
discovery.seed_hosts: ["172.16.218.199","172.16.107.151"]
cluster.initial_master_nodes: ["node-2"]
